src/components/Messages/common/MsgUpdateParamsDetails/index.tsx

```diff
@@ -0,0 +1,52 @@
+import { EncodeObject } from '@desmoslabs/desmjs';
+import { MessageDetailsComponent } from 'components/Messages/BaseMessage';
+import BaseMessageDetails, {
+  MessageDetailsField,
+} from 'components/Messages/BaseMessage/BaseMessageDetails';
+import Typography from 'components/Typography';
+import { formatCoin } from 'lib/FormatUtils';
+import React from 'react';
+import { useTranslation } from 'react-i18next';
+
+const MsgUpdateParamsDetails: MessageDetailsComponent<EncodeObject> = ({ message }) => {
+  const { t } = useTranslation('messages.common');
+  const moduleName = React.useMemo(() => {
+    const [, name] = message.typeUrl.split('.');
+    return `${name[0].toUpperCase()}${name.substring(1)}`;
+  }, [message.typeUrl]);
+
+  const fields = React.useMemo<MessageDetailsField[]>(() => {
+    const result: MessageDetailsField[] = [];
+    Object.keys(message.value.params).forEach((key) => {
+      const objectValue = message.value.params[key];
+      let serializedValue: string;
+
+      if (typeof objectValue === 'object') {
+        // Special case for the coin object.
+        if (typeof objectValue.denom === 'string' && typeof objectValue.amount === 'string') {
+          serializedValue = formatCoin(objectValue);
+        } else {
+          serializedValue = JSON.stringify(objectValue, undefined, 4);
+        }
+      } else {
+        serializedValue = objectValue.toString();
```

Comment:
Does this work on boolean values as well? I saw in your preview Gif that the `send_enabled` parameter was rendered as a white square box for some reason. 

Also, would it be better to check if the `objectValue` is either `undefined` or `null` and render it as `null` in both cases?

Reply:
@RiccardoM I have handled the case where object value is `undefined` or `null`.
About the booleans there was an error in the conversion of the `/desmos.tokenfactory.v1.MsgUpdateParams`.
I have update it to matche the [proto definition](https://github.com/desmos-labs/desmos/blob/83c3da9d96f35828c2e9b72ead2e6053941e37a9/proto/desmos/tokenfactory/v1/params.proto#L13).